### PR TITLE
[merged] Clarify srcdir != builddir bits, expand rationale

### PR DESCRIPTION
**Pull request by cgwalters**
Closes: #1

### CODE REVIEW

**Comment by cgwalters** [comment]
@cgwalters-bot delegate=mathstuf

File: build-api.md [hunk 1]
```diff
@@ -99,10 +99,14 @@ The following Build API variables may be used to declare how this
...
-* `buildapi-variable-no-builddir`: Declares that this project only works when srcdir == builddir.
+* `buildapi-variable-no-builddir`: Declares that this project only
+   works when srcdir == builddir.  Avoid this if you can - it's
+   nicest to allow users to do both.
 
 * `buildapi-variable-require-builddir`: Declares that this project
-   ONLY works when srcdir != builddir.  Avoid this if you can.
+   only works when srcdir != builddir.  Avoid this if you can, as
+   it forces build systems that are compiling for multiple architectures
+   simultaneously to create a copy of the source code.
```
Comment (mathstuf) [inline build-api.md]: Why would requiring a separate source and build directory force copies of the source tree where forced in-source builds wouldn't?

**Comment by cgwalters** [comment]
You're right, I got it backwards :smile:   Updated :arrow_up:

**Comment by mathstuf** [comment]
@cgwalters-bot r+

**Comment by cgwalters** [comment]
@cgwalters-bot retry

Had to actually finish hooking up Travis, let's see.

**Pull request closed without merging**